doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,860 @@
+            }
+
+            leaf max_flows {
+                type uint32;
+                description  "Maximum number of flow states that can be maintained per ARS object.";
+            }
+
+            leaf primary_path_threshold {
+                type uint32;
+                description  "Primary path metric";
+            }
+
+            leaf alternative_path_cost {
+                type uint32;
+                description  "Alternative path cost";
+            }
+
+            leaf-list alternative_path_members {
+                type leafref {
+                    path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                }
+                description "Members of the LAG participating in alternative path";
+            }
+        }
+        /* end of list ARS_PORTCHANNEL_LIST */
+    }
+    /* end of container ARS_PORTCHANNEL */
+
+```
+
+##### ACL_RULE changes
+
+```
+    container sonic-acl {
+
+        container ACL_RULE {
+
+            description "ACL_RULE part of config_db.json";
+
+            list ACL_RULE_LIST {
+
+                key "ACL_TABLE_NAME RULE_NAME";
+
+                leaf ACL_TABLE_NAME {
+                    type leafref {
+                        path "/acl:sonic-acl/acl:ACL_TABLE/acl:ACL_TABLE_LIST/acl:ACL_TABLE_NAME";
+                    }
+                }
+
+                leaf RULE_NAME {
+                    type string {
+                        length 1..255;
+                    }
+                } 
+...
+                leaf DISABLE_ARS_FORWARDING {
+                    description "Disable ARS forwarding on matching packets";
+                    type boolean;
+                    default false;
+                }
+            }
+         }
+    }
+```
+
+#### Config DB Enhancements  
+
+```
+; New container ARS_PROFILE_TABLE
+; ARS global configuration
+
+key                     = ARS_PROFILE|profile_name
+
+;field                  = value
+
+algorithm               = "ewma"        ;Path quality calculation algorithm
+max_flows               = uint32        ;Maximum number of flows that can be maintained for ARS
+sampling_interval       = uint32        ;Sampling interval in microseconds
+past_load_min_value     = uint16        ;Minimum value of Past load range.
+past_load_max_value     = uint16        ;Maximum value of Past load range.
+past_load_weight        = uint16        ;Weight of the past load
+future_load_min_value   = uint16        ;Minimum value of Future load range.
+future_load_max_value   = uint16        ;Maximum value of Future load range.
+future_load_weight      = uint16        ;Weight of the future load
+current_load_min_value  = uint16        ;Minimum value of Current load range.
+current_load_max_value  = uint16        ;Maximum value of Current load range.
+ipv4_enable             = boolean       ;Whether ARS is enabled over IPv4 packets

```

Comment:
both of these (ipv4/ipv6 enable) are by default turned on by TL SAI and cannot be changed to false.